code/modules/games/cards.dm

```diff
@@ -0,0 +1,454 @@
+
+/obj/item/weapon/deck/holder
+	name = "card box"
+	desc = "A small leather case to show how classy you are compared to everyone else."
+	icon_state = "card_holder"
+
+/obj/item/weapon/deck/cards
+	name = "deck of cards"
+	desc = "A simple deck of playing cards."
+	icon_state = "deck"
+
+/obj/item/weapon/deck/cards/New()
+	..()
+	var/datum/playingcard/P
+	for(var/suit in list("spades","clubs","diamonds","hearts"))
+
+		var/colour
+		if(suit == "spades" || suit == "clubs")
+			colour = "black_"
+		else
+			colour = "red_"
+
+		for(var/number in list("ace","two","three","four","five","six","seven","eight","nine","ten"))
+			P = new()
+			P.name = "[number] of [suit]"
+			P.card_icon = "[colour]num"
+			P.back_icon = "card_back"
+			cards += P
+
+		for(var/number in list("jack","queen","king"))
+			P = new()
+			P.name = "[number] of [suit]"
+			P.card_icon = "[colour]col"
+			P.back_icon = "card_back"
+			cards += P
+
+	for(var/i = 0,i<2,i++)
+		P = new()
+		P.name = "joker"
+		P.card_icon = "joker"
+		cards += P
+
+/obj/item/weapon/deck/attackby(obj/O as obj, mob/user as mob)
+	if(istype(O,/obj/item/weapon/hand))
+		var/obj/item/weapon/hand/H = O
+		if(H.parentdeck == src)
+			for(var/datum/playingcard/P in H.cards)
+				cards += P
+			qdel(H)
+			to_chat(user,"You place your cards on the bottom of \the [src].")
+			return
+		else
+			to_chat(user,"You can't mix card(s) from other deck!")
+			return
+
+	..()
+
+/obj/item/weapon/deck/attack_hand(mob/user as mob)
+	draw_card(user)
+
+/obj/item/weapon/deck/verb_pickup() // Speshul Snowflek Override for card so pickup work as intended.
+	if(!(usr)) //BS12 EDIT
+		return
+	if(!usr.canmove || usr.stat || usr.restrained() || !Adjacent(usr))
+		return
+	if((!istype(usr, /mob/living/carbon)) || (istype(usr, /mob/living/carbon/brain)))//Is humanoid, and is not a brain
+		to_chat(usr, "<span class='warning'>You can't pick things up!</span>")
+		return
+	if( usr.stat || usr.restrained() )//Is not asleep/dead and is not restrained
+		to_chat(usr, "<span class='warning'>You can't pick things up!</span>")
+		return
+	if(src.anchored) //Object isn't anchored
+		to_chat(usr, "<span class='warning'>You can't pick that up!</span>")
+		return
+	if(!usr.hand && usr.r_hand) //Right hand is not full
+		to_chat(usr, "<span class='warning'>Your right hand is full.</span>")
+		return
+	if(usr.hand && usr.l_hand) //Left hand is not full
+		to_chat(usr, "<span class='warning'>Your left hand is full.</span>")
+		return
+	if(!istype(src.loc, /turf)) //Object is on a turf
+		to_chat(usr, "<span class='warning'>You can't pick that up!</span>")
+		return
+	//All checks are done, time to pick it up!
+	usr.put_in_hands(src)
+	return
+
+/obj/item/weapon/deck/verb/draw_card(mob/user as mob)
+
+	set category = "Object"
+	set name = "Draw"
+	set desc = "Draw a card from a deck."
+	set src in view(1)
+
+	var/mob/living/carbon/human/M = user
+
+	if(usr.stat || !Adjacent(usr)) return
+
+	if(!istype(usr,/mob/living/carbon))
```

Comment:
Why are you checking carbon here if you defined user as human earlier?

Also use `user` not `usr`, as you already defined `user` in the verb parameters